docs/detections/rules-ui-create.asciidoc

```diff
@@ -263,35 +263,76 @@ For example, if a rule has an interval of 5 minutes, no additional look-back tim
 [[create-esql-rule]]
 === Create an ES|QL rule
 
-IMPORTANT: This is a placeholder for future documentation. The following content is incomplete.
+preview::[]
 
-. Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
-. To create a rule that uses ES|QL, select **ES|QL**, 
-then write a query. There are two types of ES|QL queries:
+Use {ref}/esql.html[ES|QL] to query your source events and aggregate event data. Query results are returned in a table with rows and columns. Each row becomes an alert.
+
+To create an ES|QL rule:
 
-.. **Aggregating query**: This is a query that uses the `STATS...BY` grouping commands. Query results cannot be matched with a particular document in Elasticsearch. For example:
+. Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
+. Select **ES|QL**, then write a <<esql-rule-query-types,query>>. 
 +
-[esql]
------
+TIP: Click the help icon (image:images/esql-ref-button.png[Click the ES|QL help icon,20,20]) to open the in-product reference documentation for all ES|QL commands and functions.
++
+. Click *Continue* to <<rule-ui-basic-params, configure basic rule settings>>.
+
+[float]
+[[esql-rule-query-types]]
+==== ES|QL query types 
+
+ES|QL rule queries are loosely categorized into two types: <<esql-agg-query,aggregating>> and <<esql-non-agg-query,non-aggregating>>. 
+
+[float]
+[[esql-agg-query]]
+===== Aggregating query 
+
+An aggregating query uses {ref}/esql-functions-operators.html#esql-agg-functions[`STATS...BY`] functions to aggregate source event data. Generated alerts only contain the fields returned by the query.
+
+Here is an example aggregating query:
+
+[source,esql]
+----
 FROM logs*
-| STATS count = COUNT(host.name) BY host.name
-| SORT host.name
+| STATS host_count = COUNT(host.name) BY host.name
+| SORT host_count DESC
+| WHERE host_count > 20
+----
+
+This query generates an alert whenever the `host.name` field is encountered more than 20 times during the specified rule interval.
+
+[float]
+[[esql-non-agg-query]]
+===== Non-aggregating query 
+A non-aggregating query doesn't use `STATS...BY` functions and doesn't aggregate source data. Generated alerts contain source event data and fields returned by the ES|QL query. 
+
```

Comment:
Might be useful to add an example here as well, to keep with the structure of the previous section. Would you mind verifying this as well, @vitaliidm:

```suggestion
Here is an example non-aggregating query:

[source,esql]
-----
FROM logs* [metadata _id, _index, _version]
| WHERE event.category == "process"  AND event.id == "8a4f500d"
| KEEP event.*
| LIMIT 10
-----

- This query starts by querying logs from indices that match the pattern `logs-*`. The [metadata _id, _index, _version] operator is added to turn on alert deduplication.  
- Next, the query filters events where the `event.category` is a process and the `event.id` is `8a4f500d`. 
- It keeps the selected fields `event.*`.
- Then, it limits the output to the top 10 results.
```

Reply:
I would phrase it as to allow deduplication, instead of "turn on".

Btw, this query is no valid, as it drops `_id` field. Should not it be a valid one?

Reply:
Ah yes, good catch. I'll remove that line from the query. How's this:

```suggestion
Here is an example non-aggregating query:
[source,esql]
-----
FROM logs* [metadata _id, _index, _version]
| WHERE event.category == "process"  AND event.id == "8a4f500d"
| LIMIT 10
-----
- This query starts by querying logs from indices that match the pattern `logs-*`. The `[metadata _id, _index, _version]` operator allows alert deduplication.  
- Next, the query filters events where the `event.category` is a process and the `event.id` is `8a4f500d`. 
- Then, it limits the output to the top 10 results.
```